From hackermail Thu Jan  1 00:00:00 1970
From: SeongJae Park <sj@kernel.org>
Cc: SeongJae Park <sj@kernel.org>, damon@lists.linux.dev, linux-kernel@vger.kernel.org, linux-mm@kvack.org
Message-Id: <20251208062943.68824-27-sj@kernel.org>
In-Reply-To: <20251208062943.68824-1-sj@kernel.org>
Date: Sun,  7 Dec 2025 22:29:30 -0800
Subject: [RFC PATCH v3 26/37] mm/damon: extend damon_access_report for access-origin thread info

The access reporter might be able to know which thread has made the
reporting access.  Extend the report data structure for the information.

Signed-off-by: SeongJae Park <sj@kernel.org>
Link: https://patch.msgid.link/20251208062943.68824-27-sj@kernel.org
---
 include/linux/damon.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index aff34dce7c7c..80332bb2b73c 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -116,6 +116,7 @@ struct damon_target {
  * @addr:		The start address of the accessed address range.
  * @size:		The size of the accessed address range.
  * @cpu:		The id of the CPU that made the access.
+ * @tid:		The task id of the task that made the access.
  *
  * Any DAMON API callers that notified access events can report the information
  * to DAMON using damon_report_access().  This struct contains the reporting
@@ -125,6 +126,7 @@ struct damon_access_report {
 	unsigned long addr;
 	unsigned long size;
 	unsigned int cpu;
+	pid_t tid;
 /* private: */
 	unsigned long report_jiffies;	/* when this report is made */
 };
-- 
2.47.3
